mm2src/coins/lightning.rs

```diff
@@ -712,35 +997,124 @@ impl From<ChannelDetails> for ChannelDetailsForRPC {
 }
 
 #[derive(Serialize)]
-pub struct ListChannelsResponse {
-    channels: Vec<ChannelDetailsForRPC>,
+pub struct ListOpenChannelsResponse {
+    open_channels: Vec<ChannelDetailsForRPC>,
+    limit: usize,
+    skipped: usize,
+    total: usize,
+    total_pages: usize,
+    paging_options: PagingOptionsEnum<u64>,
 }
 
-pub async fn list_channels(ctx: MmArc, req: ListChannelsRequest) -> ListChannelsResult<ListChannelsResponse> {
+pub async fn list_open_channels_by_filter(
+    ctx: MmArc,
+    req: ListOpenChannelsRequest,
+) -> ListChannelsResult<ListOpenChannelsResponse> {
     let coin = lp_coinfind_or_err(&ctx, &req.coin).await?;
     let ln_coin = match coin {
         MmCoinEnum::LightningCoin(c) => c,
         _ => return MmError::err(ListChannelsError::UnsupportedCoin(coin.ticker().to_string())),
     };
-    let channels = ln_coin
+    let mut total_open_channels: Vec<ChannelDetailsForRPC> = ln_coin
         .channel_manager
         .list_channels()
         .into_iter()
         .map(From::from)
         .collect();
 
-    Ok(ListChannelsResponse { channels })
+    total_open_channels.sort_by(|a, b| a.rpc_channel_id.cmp(&b.rpc_channel_id));
+
+    let total = total_open_channels.len();
+
+    let offset = match req.paging_options {
```

Comment:
It's typical that filters are applied *before* pagination. For example, in the GUI, you first select the filters and then get pages if filtering results do not fit.

Reply:
> It's typical that filters are applied before pagination

That's actually the case here https://github.com/KomodoPlatform/atomicDEX-API/blob/6ba33af156df9ce4984ce64e54b62326ab2f411c/mm2src/coins/lightning.rs#L1038-L1051
I think the confusion comes from setting the offset var before filtering, so maybe if I move it down the code will have better readability.

Reply:
After looking at the code I guess you are right :)
There is a problem with `rpc_id` position as I get it from the whole list, not the filtered one. Will fix it.